django/test/utils.py

```diff
@@ -841,6 +844,30 @@ def disable(self):
         setattr(Options, 'default_apps', self.old_apps)
 
 
+class time_keeper:
+    def __init__(self, timing):
+        self.timing = timing
+        self.records = []
+
+    def append(self, record, record_time):
+        new_record = '%s took %.3fs' % (record, record_time)
+        self.records.append(new_record)
+
+    @contextmanager
+    def timed(self, name):
+        start_time = time.perf_counter()
+        try:
+            yield
+        finally:
+            end_time = time.perf_counter() - start_time
+            self.append(name, end_time)
+
+    def results(self):
+        if self.timing:
```

Comment:
It seems that the `timing` parameter is doing too much work. We're storing it in the runner, plus passing it down to the time keeper class, for it only to be used here. 

Q: why is it the time keeper's job to choose whether output is displayed by the runner? (A: it's not)

I think two classes would be better than the conditional. `TimeKeeper` and `NullTimeKeeper`, then in the runner we don't store `timing` but just do: 

```
self.time_keeper = TimeKeeper() if timing else NullTimeKeeper()
```

`NullTimeKeeper` should implement no-ops for `timed()`, `append()` and `results()` and just a single `yield` for the context manager.